lib/subproviders/geth_api_double.js

```diff
@@ -19,21 +19,31 @@ function GethApiDouble(options) {
   this.initialized = false;
 
   this.initialization_error = null;
+  this.post_initialization_callbacks = [];
 
   this.state.initialize(this.options, function(err) {
     if (err) {
       self.initialization_error = err;
     }
     self.initialized = true;
+
+    var callbacks = self.post_initialization_callbacks;
+    self.post_initialization_callbacks = [];
+
+    setImmediate(function() {
+      callbacks.forEach(function(callback) {
+        callback();
+      });
+    });
   });
 }
 
 GethApiDouble.prototype.waitForInitialization = function(callback) {
   var self = this;
   if (this.initialized == false) {
-    setTimeout(function() {
+    this.post_initialization_callbacks.push(function() {
       self.waitForInitialization(callback);
-    }, 100);
+    });
```

Comment:
Why `push` an anonymous function w/ a call to `waitForInitialization` here? Why not just `push` the `callback` itself?

Reply:
I've kept this similar to the previous behavior. Note how in the else condition the error and state are provided to `callback`. Pushing `callback` itself means that has to be replicated in the `initialize` callback above.

Reply:
I think it'd be good to simplify the control flow a bit by making the functions pushed to `post_initialization_callbacks` have the same signature (`function(err, result)`), as this is a pattern used for callbacks most other places in the code.

If you have the time to do that, I'd appreciate it. Otherwise I'll likely refactor this myself but after I finish up some more pressing changes.

Reply:
Sure, I just went for the fewest possible changes. 

I filed this PR on behest of a client, but sadly that contract has finished. I won't have time to follow-up on this, so please go ahead and use this as you will 😄